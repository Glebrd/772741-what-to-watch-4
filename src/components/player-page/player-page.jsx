import React from "react";
import withVideo from "../../hocs/with-video/with-video";
import PlayerControls from "../player-controls/player-controls.jsx";
import PropTypes from "prop-types";
import {movieType} from "../../types";
import history from "../../history";
import {getCurrentMovie, getCurrentMovieByID} from "../../reducer/application/selectors";
import {connect} from "react-redux";
import {getMovies} from "../../reducer/data/selectors";
import {Link} from "react-router-dom";

const getSvgIconPlayPause = (isPlaying) =>
  isPlaying
    ? `#pause`
    : `#play-s`;

const PlayerPage = (props)=>{
  const {currentMovie, videoRef, onPlayPause, onFullScreen, isPlaying, history} = props;
  const {picture, videoLink, title} = currentMovie;
  return (
    <div className="player">
      <video
        src={videoLink}
        ref={videoRef}
        className="player__video" poster={picture}
      >
      </video>


      <button
        onClick={history.goBack}
        type="button" className="player__exit">Exit </button>


      <div className="player__controls">

        <PlayerControls
          videoRef={videoRef}
        />

        <div className="player__controls-row">
          <button
            onClick={onPlayPause}
            type="button" className="player__play">
            <svg viewBox="0 0 19 19" width="19" height="19">
              <use xlinkHref={getSvgIconPlayPause(isPlaying)}
              ></use>
            </svg>
            <span>Play</span>
          </button>
          <div className="player__name">{title}</div>

          <button
            onClick={onFullScreen}
            type="button" className="player__full-screen">
            <svg viewBox="0 0 27 27" width="27" height="27">
              <use xlinkHref="#full-screen"></use>
            </svg>
            <span>Full screen</span>
          </button>
        </div>
      </div>
    </div>
  );
};

PlayerPage.propTypes = {
  currentMovie: movieType,
  videoRef: PropTypes.object,
  onPlayPause: PropTypes.func,
  onFullScreen: PropTypes.func,
  isPlaying: PropTypes.bool,
  onExit: PropTypes.func,
};


const mapStateToProps = (state, props) => ({
  currentMovie: getCurrentMovieByID(state, props.match.params.id),
});

export {PlayerPage};

export default connect(mapStateToProps)(withVideo(PlayerPage));
