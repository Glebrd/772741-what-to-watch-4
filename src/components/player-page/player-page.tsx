import * as React from "react";
import withVideo from "../../hocs/with-video/with-video";
import PlayerControls from "../player-controls/player-controls";
import {getCurrentMovieByID} from "../../reducer/application/selectors";
import {connect} from "react-redux";
import history from "../../history";
import {MovieType} from "../../types";

const getSvgIconPlayPause = (isPlaying) =>
  isPlaying
    ? `#pause`
    : `#play-s`;

interface Props {
  currentMovie: MovieType;
  videoRef: React.RefObject<HTMLVideoElement>;
  onPlayPause: () => {void};
  onFullScreen: () => {void};
  isPlaying: boolean;
  onExit: () => {void};
}

const PlayerPage: React.FunctionComponent<Props> = (props: Props) => {
  const {currentMovie, videoRef, onPlayPause, onFullScreen, isPlaying} = props;
  const {picture, videoLink, title} = currentMovie;
  return (
    <div className="player">
      <video
        src={videoLink}
        ref={videoRef}
        className="player__video" poster={picture}
      >
      </video>


      <button
        onClick={history.goBack}
        type="button" className="player__exit">Exit </button>


      <div className="player__controls">

        <PlayerControls
          videoRef={videoRef}
        />

        <div className="player__controls-row">
          <button
            onClick={onPlayPause}
            type="button" className="player__play">
            <svg viewBox="0 0 19 19" width="19" height="19">
              <use xlinkHref={getSvgIconPlayPause(isPlaying)}
              ></use>
            </svg>
            <span>Play</span>
          </button>
          <div className="player__name">{title}</div>

          <button
            onClick={onFullScreen}
            type="button" className="player__full-screen">
            <svg viewBox="0 0 27 27" width="27" height="27">
              <use xlinkHref="#full-screen"></use>
            </svg>
            <span>Full screen</span>
          </button>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state, props) => ({
  currentMovie: getCurrentMovieByID(state, props.match.params.id),
});

export {PlayerPage};

export default connect(mapStateToProps)(withVideo(PlayerPage));
