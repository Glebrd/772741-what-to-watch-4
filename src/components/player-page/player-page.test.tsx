import * as React from "react";
import * as renderer from "react-test-renderer";
import {PlayerPage} from "./player-page";
import {Router} from "react-router-dom";
import history from "../../history";
import noOperation from "../../const";

const currentMovie = {
  id: 7,
  title: `Revenant `,
  picture: `img/revenant.jpg`,
  genre: `Comedy`,
  date: 2015,
  poster: `img/revenant.jpg`,
  background: `img/revenant.jpg`,
  rating: 8.1,
  scores: 556,
  director: `Nisse Prahm`,
  starring: [`Eirena Corain`, `Kamila Skelhorn`, `Manuel Elcott`],
  description: `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
  videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  runTime: 120,
  backgroundColor: `#D8E3E5`,
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  isFavorite: true,
};

jest.mock(`../player-controls/player-controls`, () => {
  return {
    'default': `PlayerControls`
  };
});

it(`Player page renders correctly`, () => {
  const isPlaying = false;
  const tree = renderer
    .create(
        <Router history={history}>
          <PlayerPage
            onExit={noOperation}
            currentMovie={currentMovie}
            videoRef={React.createRef()}
            isPlaying={isPlaying }
            onFullScreen={noOperation}
            onPlayPause={noOperation}
          />
        </Router>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
