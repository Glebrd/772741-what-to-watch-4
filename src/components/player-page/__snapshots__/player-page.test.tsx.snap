// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player page renders correctly 1`] = `
<div
  className="player"
>
  <video
    autoPlay={false}
    className="player__video"
    muted={true}
    poster="img/revenant.jpg"
    src="http://media.xiph.org/mango/tears_of_steel_1080p.webm"
  />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit 
  </button>
  <div
    className="player__controls"
  >
    <PlayerControls
      videoRef={
        Object {
          "current": null,
        }
      }
    />
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Revenant 
      </div>
      <button
        className="player__full-screen"
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
